src/stream_inserter.cpp

```diff
@@ -14,13 +14,112 @@
 
 #include "stream_inserter.h"
 #include <memory>
-#include "grpcpp/client_context.h"
+#include <grpcpp/client_context.h>
 
 namespace greptime {
     
-using greptime::v1::RequestHeader;
+auto to_insert_requests(std::vector<InsertRequest>& vec_insert_requests) -> InsertRequests {
+    InsertRequests insert_requests;
+    for (auto &insert_request : vec_insert_requests) {
+        insert_requests.add_inserts()->Swap(&insert_request);
+    }
+    return std::move(insert_requests);
+}
+
+uint32_t InsertEntry::get_point_num() {
+    uint32_t ret = 0;
+    for (const auto & table : tables) 
+        ret += table.second.size();
+    return ret;
+}
+
+void LineWriter::add_row(TableName table_name, Timestamp ts, const std::unordered_map<FieldName, FieldVal> &fields) {
```

Comment:
As per implementation, I'd suggest to add a read-write lock to protect `LineWriter::mp` and minimize the lock contention.